.github/workflows/run-unit-tests.yml

```diff
@@ -0,0 +1,76 @@
+name: UnitTests
+
+on:
+  workflow_call:
+    inputs:
+      dotnet_sdk_version:
+        required: true
+        type: string
+      npm_package_source:
+        required: false
+        type: string
+    secrets:
+      npm_auth_token:
+        required: false
+      myget_auth_token:
+        required: false
+      myget_user:
+        required: false
+      myget_pass:
+        required: false
+
+jobs:
+  UnitTests:
+    runs-on: ubuntu-latest
+    concurrency:
+      group: test-stage
+      cancel-in-progress: true
+    steps:
+      - uses: actions/checkout@v3
+
+      - uses: actions/cache@v3
+        id: cache-nuget
+        with:
+          path:  ${{ github.workspace }}/.nuget/packages
+          key: ${{ runner.os }}-nuget-${{ hashFiles('**/*.csproj') }}
+          restore-keys: |
+            ${{ runner.os }}-nuget-
+
+      - name: 🔧 Setup nuget
+        uses: nuget/setup-nuget@v1
+
+      - name: 🔧 Setup .NET SDK
+        uses: actions/setup-dotnet@v3
+        with:
+          dotnet-version: ${{ inputs.dotnet_sdk_version }}
+
+      - name: 🗝️ Authenticate npm registry
+        run: npm config set ${{ inputs.npm_package_source }}:_authToken ${{ secrets.npm_auth_token }}
+        if: inputs.npm_package_source != ''
+
+      - name: "🗝️ Authenticate MyGet"
+        uses: moryx-industry/nuget-setapikey@main
+        with:
+          sources-file: './myget.feeds'
+          api-key: ${{ secrets.myget_auth_token }}
+          registry-user: ${{ secrets.myget_user }}
+          registry-pass: ${{ secrets.myget_pass }}
+
+      - name: ♺ Execute nuget restore
+        run: nuget restore
+          
+      - name: 🧰 Setup BuildToolkit
+        uses: moryx-industry/setup-buildtoolkit@main
+
+      - name: 🧪 Run tests 
+        run: |
+          .\unit-tests.ps1
```

Comment:
where does this ps script come from? and how does it differentiate integration and units tests, or doesn't it deifferentiate at all? 🙈

Reply:
Its content has been moved from `build.ps1`. I thought it would be cleaner to do `.\unit-tests.ps1 -with parameters` instead of `.build.ps1 -UnitTests -with parameters`.
There's no differentiation right now. I'll change that workflow to `run-tests` and add a filter parameter.